protos/flyteidl/core/resource.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package flyteidl.core;
+
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core";
+
+import "google/protobuf/timestamp.proto";
+
+// Cluster resource that task and workflow will run on
+message Resource {
+  oneof value {
+    RayCluster ray = 1;
+  }
+}
+
+
+// Define Ray cluster spec
+message RayCluster {
```

Comment:
should we move the Ray specific messages into a separate file?  If not, should we add "Ray" to the names?  I know renaming is annoying cuz of all the generated/model code you have to